props/saber_sabersense_buttons.h

```diff
@@ -532,7 +564,15 @@ class SabersenseButtons : public PROP_INHERIT_PREFIX PropBase {
       SaberBase::DoNewFont();  // Play font ident if 'array' sound file missing.
     }
 #endif
+#ifndef SABERSENSE_DISABLE_SAVE_ARRAY
+    SaveArrayState();  // After changing array, save new array index.
+  }
```

Comment:
This is super messy. Please have your #if #else #endif stay within one function
(or wrap an entire function.)